### ISSUE TITLE
get_cmdline_args() does not include --headless

### ISSUE THREAD

**Issue by mweibel** [godotengine/godot#84019]
### Godot version

4.1.2.stable

### System information

Godot v4.1.2.stable - Windows 10.0.19045 - Vulkan (Forward+)

### Issue description

[Docs about Starting the dedicated server](https://docs.godotengine.org/en/stable/tutorials/export/exporting_for_dedicated_servers.html#starting-the-dedicated-server) mention you can check for headless using `if "--headless" in OS.get_cmdline_args():`. 

However, when trying this I noticed `--headless` is not in `get_cmdline_args()`:
```
 & 'C:\Program Files\Godot\Godot_v4.1.2-stable_win64.exe' -d --headless --path .
PS C:\Users\Michael\code\games\godot-racing> Godot Engine v4.1.2.stable.official.399c9dc39 - https://godotengine.org

DisplayServer.get_name(): headless
OS.get_cmdline_args(): []
```

It also doesn't output it when using `Godot_v4.1.2-stable_win64_console.exe`:

```
 & 'C:\Program Files\Godot\Godot_v4.1.2-stable_win64_console.exe' -d --headless --path .
Godot Engine v4.1.2.stable.official.399c9dc39 - https://godotengine.org
 CC
DisplayServer.get_name(): headless
OS.get_cmdline_args(): []
```

### Steps to reproduce

1. create a new project with a node containing the following script:
```
extends Node


# Called when the node enters the scene tree for the first time.
func _ready():
	print("DisplayServer.get_name(): ", DisplayServer.get_name())
	print("OS.get_cmdline_args(): ", OS.get_cmdline_args())
```

2. save the scene
3. define the scene as main scene & save project
4. run the project using godot (make sure to adjust the path):
```
 & 'C:\Program Files\Godot\Godot_v4.1.2-stable_win64.exe' -d --headless --path .
```

### Minimal reproduction project

[cmdline_args.zip](https://github.com/godotengine/godot/files/13181685/cmdline_args.zip)

**Comment by Calinou** [comment]
There are a bunch of engine arguments like this that get trimmed from `get_cmdline_args()`'s output from some reason, but I could nevere understand why.

**Comment by mweibel** [comment]
@Calinou yeah that's what I switched to. If this is the "blessed" way, I can make a PR to adjust the documentation for the mentioned page accordingly.

**Comment by Calinou** [comment]
> If this is the "blessed" way, I can make a PR to adjust the documentation for the mentioned page accordingly.

Sure, go ahead :slightly_smiling_face: 

In general, that approach is also more robust as it'll work if you build a binary that only has the headless display server.

**Comment by mweibel** [comment]
Yeah makes sense.

docs change is in https://github.com/godotengine/godot-docs/pull/8362

**Comment by Calinou** [comment]
Reopening, as there's an underlying issue that may be a bug.

**Comment by Praytic** [comment]
It was already merged. What bug are you referring to? @Calinou

**Comment by AThousandShips** [comment]
Was reopened despite the merged PR as this might be a bug in itself, that the missing argument is a bug as opposed to the check being wrong 🙂
